### ISSUE TITLE
alpha precision

### ISSUE THREAD

**Issue by ferllings** [Qix-/color#174]
Hello,

When I parse hex values with alpha #000000aa
I get { model: 'rgb', color: [ 0, 0, 0 ], valpha: 0.67 }
The same valpha value is calculated for #000000ab

This is probably due to the rounding when converting from 255 base to 1 base.
Is there a way to access the intermediate 255 base value?

I understand the difference isn't much between both values, but the problem is that I can't retrieve the original hex value from the object... both giving #000000aa

Any idea?

**Comment by Qix-** [comment]
I don't quite understand what you're asking - can you show some code with expected output and what you're really getting?

**Comment by ferllings** [comment]
```
Color("#000000aa") returns
{ model: 'rgb', color: [ 0, 0, 0 ], valpha: 0.67 }
```

```
Color("#000000ab") returns
{ model: 'rgb', color: [ 0, 0, 0 ], valpha: 0.67 }
```

So 2 hexa values produce the same object

And of course when I'm trying to reconstitute the alpha value using the object I get the same result for both. "aa"

**Comment by Qix-** [comment]
Wow, this is definitely a bug.

**Comment by Qix-** [comment]
Okay so digging deep into `color-string`'s history, this was a bug that was introduced as a solution to another bug.

- Hex support wasn't added until Qix-/color-string#33
- A bug was uncaught there that rounded the alpha to either 0 or 1, which was "fixed" in https://github.com/Qix-/color-string/commit/eb857c2e5f421373a00057d08afb34f45b1095e1
- However, the *real* fix should have been simply to remove the `Math.round`. Not sure what the rationale was to lock it to two digit precision (perhaps for aesthetic? which I'm over now). But there was no technical reason for it that I can find, and the other string parsers do not make the same guarantee.

I'll fix this in `color-string` and release shortly. Apologies for the delay, the pandemic made life a lot harder.

**Comment by Qix-** [comment]
Released as `3.1.3`, thanks again for the report and apologies for how long it took to address.

**Comment by ferllings** [comment]
Thanks a lot. 
And no worries, I know what it's like to work on opensource projects.
